source/kernel/src/drivers/emb_display.rs

```diff
@@ -0,0 +1,260 @@
+use crate::{
+    comms::{
+        kchannel::{KChannel, KConsumer},
+        oneshot::Reusable,
+    },
+    registry::{
+        Envelope, KernelHandle, Message, RegisteredDriver, ReplyTo,
+    },
+    Kernel,
+};
+use maitake::sync::Mutex;
+use mnemos_alloc::containers::{HeapArc, HeapArray, HeapFixedVec};
+// use tracing::{debug, warn};
+use uuid::Uuid;
+use embedded_graphics::{
+    pixelcolor::{Gray8, GrayColor},
+    prelude::*,
+    image::ImageRaw,
+};
+
+const BYTES_PER_PIXEL: u32 = 1;
+const DISP_WIDTH: u32 = 319;
+const DISP_HEIGHT: u32 = 239;
+
+// Registered driver
+pub struct EmbDisplay {
+    kernel: &'static Kernel,
+}
+
+// FrameChunk is recieved after client has sent a request for one
+pub struct FrameChunk {
+    frame_id: u16,
+    bytes: HeapArray<u8>,
+    start_x: i32,
+    start_y: i32,
+    width: u32,
+    height: u32,
+}
+
+struct FrameInfo {
+    frame: u16,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum FrameError {
+    DuplicateItem,
+    RegistryFull,
+}
+
+struct DisplayInfo{
+    kernel: &'static Kernel,
+    frames: HeapFixedVec<FrameInfo>,
+    frame_size: usize,
+}
+
+// Client interface to EmbDisplay
+pub struct EmbDisplayHandle {
+    prod: KernelHandle<EmbDisplay>,
+    reply: Reusable<Envelope<Result<Response, FrameError>>>,
+}
+
+pub enum Request {
+    NewFrameChunk { frame_id: u16, start_x: i32, start_y: i32, width: u32, height: u32 },
+}
+
+pub enum Response {
+    FrameChunkAllocated(FrameChunk),
+}
+
+pub enum EmbDisplayError {
+    BufferFull,
+}
+
+struct CommanderTask {
+    cmd: KConsumer<Message<EmbDisplay>>,
+    fmutex: HeapArc<Mutex<DisplayInfo>>, 
+}
+
+// impl EmbDisplay
+impl RegisteredDriver for EmbDisplay {
+    type Request = Request;
+    type Response = Response;
+    type Error = FrameError;
+    const UUID: Uuid = crate::registry::known_uuids::kernel::EMB_DISPLAY;
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum RegistrationError {
+    DisplayAlreadyExists,
+}
+
+// Register the driver instance
+impl EmbDisplay {
```

Comment:
nit: i think this comment was supposed to go on the method?
```suggestion
impl EmbDisplay {
    /// Register the driver instance
```

Reply:
makes sense!